db/view/view.cc

```diff
@@ -3119,6 +3112,9 @@ class view_builder::consumer {
     build_step& _step;
     built_views _built_views;
     gc_clock::time_point _now;
+    // Base schema should be updated from the step's base schema
+    // at the same time we update the views to build.
+    schema_ptr _base_schema;
```

Comment:
I'm afraid I didn't understand how _base_schema is different from _base.schema. Isn't the whole point that we always update _base when we update the view?